src/geo/io.clj

```diff
@@ -21,8 +23,38 @@
 (defn read-wkt [^String wkt] (.read wkt-reader wkt))
 (defn to-wkt [^Geometry geom] (.write wkt-writer geom))
 
-(defn read-wkb [^bytes bytes] (.read wkb-reader bytes))
-(defn to-wkb [^Geometry geom] (.write wkb-writer geom))
+(defn read-wkb
+  "Read a WKB byte array and convert to a Geometry"
+  [^bytes bytes]
+  (.read wkb-reader bytes))
+
+(defn read-wkb-hex
+  "Read a WKB hex string and convert to a Geometry"
+  [^String s]
+  (read-wkb (WKBReader/hexToBytes s)))
+
+(defn to-wkb
+  "Write a WKB, excluding any SRID"
+  [^Geometry geom]
+  (.write wkb-writer geom))
+
+(defn to-ewkb [^Geometry geom]
+  "Write an EWKB, including the SRID"
+  (let [dims (jts/geom-dimension-check geom)]
```

Comment:
I think JTS has a method for this: https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#getDimension()

Is it possible to use that here? In general I'd like to avoid anything that requires traversing all the coordinates of a geometry

Reply:
Yes, I wasn't super pleased to end up having to pass through the coordinates like that. I was originally trying to use .getDimension but I think that's something different -- the javadoc states that "Returns the dimension of this geometry. The dimension of a geometry is is the topological dimension of its embedding in the 2-D Euclidean plane. In the JTS spatial model, dimension values are in the set {0,1,2}.

Note that this is a different concept to the dimension of the vertex Coordinates. The geometry dimension can never be greater than the coordinate dimension. For example, a 0-dimensional geometry (e.g. a Point) may have a coordinate dimension of 3 (X,Y,Z)."

I don't know of any function that can determine that "coordinate dimension", but I'll look a little more closely. It'd be nice to speed that up.